lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -310,6 +298,15 @@ def initconfig_hardware_entries(self):
                     )
                 )
 
+            # Extra definitions that might be required in other cases
+            if not spec.satisfies("^blt"):
+                entries.append(cmake_cache_path("HIP_ROOT_DIR", "{0}".format(spec["hip"].prefix)))
+
+                if archs[0] != "none":
+                    arch_str = ";".join(archs)
+                    entries.append(cmake_cache_string("AMDGPU_TARGETS", arch_str))
+                    entries.append(cmake_cache_string("GPU_TARGETS", arch_str))
```

Comment:
Did something fail w/o these? because my opinion if these are project specific options they should go in the project. But i am fine with the consensus.

Reply:
Nothing failed without these but I haven't run the tests for non-radiuss projects.
Even RAJA for example sets these in it's lc-builds scripts, and we already had the conversation in the past, the conclusion being: we are afraid to remove them.

As a first step, I'm removing them for radiuss projects with ^blt.

I like your idea to let projects add that to their build system if they need it.
Maybe an intermediate step would be to move those extra definitions to the projects package instead of keeping them in the CachedCMakePackage.